crates/trie/sparse-parallel/src/trie.rs

```diff
@@ -2409,19 +2402,17 @@ impl SparseSubtrieInner {
                         );
                         update_actions
                             .push(SparseTrieUpdatesAction::InsertUpdated(path, branch_node));
-                    } else if branch_node_tree_masks.get(&path).is_some_and(|mask| !mask.is_empty()) ||
-                        branch_node_hash_masks.get(&path).is_some_and(|mask| !mask.is_empty())
-                    {
-                        // If new tree and hash masks are empty, but previously they weren't, we
-                        // need to remove the node update and add the node itself to the list of
-                        // removed nodes.
-                        update_actions.push(SparseTrieUpdatesAction::InsertRemoved(path));
-                    } else if branch_node_tree_masks.get(&path).is_none_or(|mask| mask.is_empty()) &&
-                        branch_node_hash_masks.get(&path).is_none_or(|mask| mask.is_empty())
-                    {
-                        // If new tree and hash masks are empty, and they were previously empty
-                        // as well, we need to remove the node update.
-                        update_actions.push(SparseTrieUpdatesAction::RemoveUpdated(path));
+                    } else {
```

Comment:
the logic here is equivalent where is bascially just   "Did this path previously have non-empty masks?"

```
  if (previously had non-empty masks) {
      InsertRemoved    // was in DB, now gone → mark as removed
  } else {
      RemoveUpdated    // wasn't in DB anyway → just remove the update
  }

```